api.js

```diff
@@ -258,17 +259,42 @@ Api.prototype.mediaPut = function (req, res, m, q) {
     return
   }
 
-  var id = randombytes(16).toString('hex')
+  var self = this
+
+  var ext = path.extname(q.file)
+  var id = randombytes(16).toString('hex') + ext
   res.setHeader('content-type', 'application/json')
 
-  fs.createReadStream(q.file).pipe(this.media.createWriteStream(id))
-    .once('finish', function () {
-      res.end(JSON.stringify({id: id}))
-    })
-    .once('error', function (err) {
+  var mediaPath = 'original/' + id
+  var thumbnailPath = 'thumbnail/' + id
+
+  function copyFileTo (file, to, cb) {
+    fs.createReadStream(file).pipe(self.media.createWriteStream(to))
+      .once('finish', cb)
+      .once('error', cb)
+  }
+
+  copyFileTo(q.file, mediaPath, function (err) {
+    if (err) {
       res.statusCode = 500
       res.end(err.toString())
+      return
+    }
+
+    if (!q.thumbnail) {
```

Comment:
@karissa and I briefly discussed this. I think for now we should make q.thumbnail a required parameter.

Reply:
Why?